Upgrade to SDK v3.4

mody opened the pull request:
Also works with the old SDK

BFW-5632

mody committed:
Update Dockerfile to provide SDK v3.4

BFW-5632

mody committed:
Fix uart_nic to work with SDK v3.4

BFW-5632

mody committed:
Update sdkconfig; disable color logs

BFW-5632

mody committed:
Update custom initialization according the SDK

BFW-5632

Tomcus commented on Dockerfile:
```diff
@@ -3,14 +3,10 @@ FROM ubuntu:22.04
 
-RUN apt-get update
-RUN apt-get -y install bash git cmake python3 python3-pip python3-virtualenv python3-cryptography python3-future python3-click python3-serial python3-wheel python3-pyparsing python3-pyelftools
+RUN apt-get update && apt-get -y install bash git cmake python3 python3-pip python3-virtualenv python3-cryptography python3-future python3-click python3-serial python3-wheel python3-pyparsing python3-pyelftools python-is-python3 libncurses-dev flex bison gperf
```
Just a nitpick, but I think this long script is bit unreadable. Can you please do the `&& \<newline>` to format it better.

mody replied:
Fixed in c2d03d9b8b299ef0c71d9928c87b7066ebb20e23

Tomcus commented on CMakeLists.txt:
```diff
@@ -2,6 +2,7 @@
 # CMakeLists in this exact order for cmake to work correctly
 cmake_minimum_required(VERSION 3.5)
 
+set(LANGUAGE, C)
```
Do we need specify the language here? I would like to have some copied files from the main printer repo to make sure they are 1:1 and I would like to write them in c++.

mody replied:
Fixed in 6945f3a06adbb5a5fbb37f87395c202c4c687915

Tomcus reviewed (approved):
LGTM

mody committed:
Fix error "wifi: err1,exceed max time value,max val is 000689D0"

The timeout was too large and driver complained.
The biggest value not failing is 400.

BFW-5632
